content/docs/user-guide/managing-external-data.md

```diff
@@ -2,12 +2,13 @@
 
 > ⚠️ This is an advanced feature for very specific situations and not
 > recommended except if there's absolutely no other alternative. In most cases
-> alternatives like the
-> [to-cache](/doc/command-reference/add#example-transfer-to-the-cache) or
-> [to-remote](/doc/command-reference/add#example-transfer-to-remote-storage)
-> strategies of `dvc add` and `dvc import-url` are more convenient. **Note**
-> that external outputs are not pushed or pulled from/to
-> [remote storage](/doc/command-reference/remote).
+> alternatives like the [to-cache] or [to-remote] strategies of `dvc add` and
+> `dvc import-url` are more convenient. **Note** that external outputs are not
+> pushed or pulled from/to [remote storage].
+
+[to-cache]: /doc/command-reference/add#example-transfer-to-the-cache
+[to-remote]: /doc/command-reference/add#example-transfer-to-remote-storage
+[remote storage]: /doc/command-reference/remote
```

Comment:
WDYT @habernal ? Much prettier MD code (cc @iesahin)

The first 2 are refactored because they use an `#anchor`. The last one because it repeats later (line 43).

Reply:
Oops meant @shcheklein

Reply:
I'm fine with this. Looks nice! But we need to do this only when we make sure that link checker works to my mind. wdyt?

Reply:
> i.e. where is the link checker is running now? Not seeing it in the PR checks.

Ah OK the deployment was outdated. I redeployed it and the link checker checks triggered. A bit confused as to why there's 2 but OK, checking...

Reply:
Ah found https://github.com/iterative/link-check.action too...

Reply:
@rogermparent this keeps getting

`There was an error trying to get a merge-base! Falling back to "origin/master"`

so I can't really tell whether it even works with this kind of MD link.

Do you know why that happens? See [check](https://github.com/iterative/dvc.org/pull/2534/checks?check_run_id=2806436984). Thanks

Reply:
The action's code seems totally obfuscated so I don't know how to debug ☝️ or determine whether any md link style is supported. Created https://github.com/iterative/link-check.action/issues/1 and https://github.com/iterative/link-check.action/issues/2 for now.

Reply:
> The action's code seems totally obfuscated so I don't know how to debug

The source repo is at https://github.com/iterative/link-check, GHA uses GitHub repos to deploy, and since the application needs to be transpiled/bundled I had created the `.action` repo to serve as the deploy bed without shoving in the source history. I recently added [a bit to the README](https://github.com/iterative/link-check.action#contributing) clarifying this, but maybe it could be a bit higher up.

Reply:
> `There was an error trying to get a merge-base! Falling back to "origin/master"`

I'm pretty confident this is unrelated to the new link types, and also doesn't block the check. The diff the action checks against is generated by doing a two-step `merge-base ${ancestor} HEAD` and `diff ${base} HEAD`, like GitHub diffs do to take into account changes merged into master since the PR's creation.

If `merge-base` fails for some reason, the actions logs that it happened and the diff falls back to a simple `diff ${ancestor} HEAD` (in this case, `ancestor` is `origin/master`) and checks against that. The warning here is more of a red herring, since the actual problem is the link checker simply not seeing these different styles of links.

To account for diffs, the link checker uses hand-rolled logic with different sets of regexes for each file extension. I know I wasn't aware of these alternate link styles, so if the link checker did handle them it would be entirely coincidence. The logic can, however, be easily extended to account for these links as long as we can programmatically find them.

The source for the logic that does this can be found in [the `scrapeLinks` module](https://github.com/iterative/link-check/blob/23f1a5ef61b682746599236da6ca700260517489/src/scrapeLinks.ts#L18).

Reply:
Ah thanks for the source repo.

> doesn't block the check

I see. I thought the check was dying there but I guess then we just don't get a log of what the check did except for warnings like that one or errors (e.g. dead links).

> the link checker simply not seeing these different styles of links

OK well that's what I was trying to confirm.

> be easily extended to account for these links as long as we can programmatically find them.
The source for the logic that does this can be found in the scrapeLinks module.

I'll give it a look. For now I transferred over https://github.com/iterative/link-check/issues/4 and this PR or similar future changes depend on that now.

p.s. how do I bundle/deploy the action once modified? See https://github.com/iterative/link-check/issues/5 thanks!